plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/OpenShiftServerExtendedProperties.java

```diff
@@ -42,16 +43,31 @@ public boolean hasWelcomePage() {
 	}
 
 	@Override
-	public String getWelcomePageUrl() {
+	public String getWelcomePageUrl() throws GetWelcomePageURLException {
 		String welcomePageUrl = null;
-		IService service = OpenShiftServerUtils.getService(server);
+
+		//Get connection explicitly to report failure. Try and connect right now to know if it fails.
+		//Do not catch OpenShiftException, let it be reported. We are more concerned of NPE.
+		Connection connection = OpenShiftServerUtils.getConnection(server);
+		if(connection == null || !connection.connect()) {
+			throw new GetWelcomePageURLException("Connection is not established.");
+		}
+
+		IService service = OpenShiftServerUtils.getService(server, connection);
+		if(service == null) {
+			throw new GetWelcomePageURLException("Service is missing.");
+		}
+
 		IProject project = service.getProject();
```

Comment:
Not specific to your pull, but there is no need to get a project to get the routes.  You can accomplish this same call using the connection and retrieving the required project information by service#getNamespace().  This additionally saves you a REST call to load the project resource, and you could remove the throw from the null project all together.